src/module.c

```diff
@@ -5853,7 +5853,17 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
     }
 
     if (flags & REDISMODULE_ARGV_RESPECT_DENY_OOM) {
-        if (cmd->flags & CMD_DENYOOM) {
+        uint64_t flags = cmd->flags;
+
+        if (cmd->group == COMMAND_GROUP_SCRIPTING) {
+            if (cmd->proc == fcallCommand) { /* FCALL */
```

Comment:
probably can just work off of cmd->proc in general, no need for previous if.

the reason for now _ro functions is that they should be allowed in OOM situation by default I'd think?

Reply:
ok, if that code was only about OOM, but now it's not...

i still think it'll be better to extract that ugly `if` in server.c to a dedicated function to reuse it here.

Reply:
yes, already changed it per that.  no longer just OOM, now uses the cmd flags built from the script flags in all places.

Reply:
re ugly code, I guess you mean this?

```    uint64_t cmd_flags = c->cmd->flags;
    if (c->cmd->proc == evalCommand || c->cmd->proc == evalShaCommand ||
        c->cmd->proc == evalRoCommand || c->cmd->proc == evalShaRoCommand ||
        c->cmd->proc == fcallCommand || c->cmd->proc == fcallroCommand)
    {
        if (c->cmd->proc == fcallCommand || c->cmd->proc == fcallroCommand)
            cmd_flags = fcallGetCommandFlags(c, cmd_flags);
        else
            cmd_flags = evalGetCommandFlags(c, cmd_flags);
    }
    ```

Reply:
yes, it's very specific code to handle special concerns of several commands, in a place that's suppose to be rather generic.
i'd rather put it in a specific function rather than clone it.

Reply:
ok, something like this

```
uint64_t getCommandFlags(client *c) {
    uint64_t cmd_flags = c->cmd->flags;
    
    if (c->cmd->proc == fcallCommand || c->cmd->proc == fcallroCommand) {
        cmd_flags = fcallGetCommandFlags(c, cmd_flags);
    } else if (c->cmd->proc == evalCommand || c->cmd->proc == evalRoCommand ||
               c->cmd->proc == evalShaCommand || c->cmd->proc == evalShaRoCommand) { /* EVAL* */
        cmd_flags = evalGetCommandFlags(c, cmd_flags);
    }
    
    return cmd_flags;
}```